### DISCUSSION TITLE
XAML Designer rise an exception while creating ChromiumWebBrowser in XAML

### DISCUSSION THREAD

**Discussion by DrCQ** [cefsharp/CefSharp#4541] [General]
I use CefSharp (V114.2.100) to create an _ChromiumWebBrowser_ instance in the XAML code. The execution of the application has no errors, but the XAML Designer (VS 17.6.4) is rising an exception:
_**Parameter "parameter" (object) cannot be of type Microsoft.VisualStudio.XSurface.Wpf.Window, as the command type requires an argument of type System.Windows.Window. (Parameter 'parameter')**_

I contacted Microsoft Support but without proper response yet!

The XAML code causing this exception is part of a Button binding command:
`CommandParameter="{Binding Mode=OneWay, RelativeSource={RelativeSource AncestorType={x:Type local:MainWindow}}}"`

During my investigation, I found the following hint from VS: _Cannot load a reference assembly for execution_

Any idea why the XAML Designer cannot find the proper assembly, while the execution (Debug and Release mode) is running without any problems?
You can find the demo application [here](https://github.com/DrCQ/Example_NET_WPF_Plain) with only few lines of XAML and VB code.
Regards,

Christoph

**Comment by amaitland** [comment]
For background `CefSharp` uses `VC++`.  Specifically `CefSharp.Core.Runtime.dll` is a [mixed mode](https://learn.microsoft.com/en-us/cpp/dotnet/mixed-native-and-managed-assemblies?view=msvc-170) C++/CLI assembly. In the `NETCore` package there is a `ref` assembly which is used for compilation with the the actual arch specific dll being used at runtime.

The package structure is described at https://github.com/cefsharp/CefSharp/issues/2796#issuecomment-618874991

With the old `.Net Framework` designer we used to specifically avoid loading any reference to `CefSharp.Core.Runtime.dll` when in design mode. I've not spent any time looking into how `NetCore` designers work for `WPF`. I know `WinForms` added a new `OOP` designer to workaround the issue of `VS` being written in `.net 4.8` and subsequently being unable to load a `NetCore` assembly. Did WPF do something similar?

Does the `WPF` designer have it's own output window in VS? There might be some more detail as to what's going on there.

> I contacted Microsoft Support but without proper response yet!

MS are likely the best qualified to diagnose/fix the issue. At best this would be something we'd have to workaround. The project provides a `ref` assembly which should be used in place of the arch specific assembly. If that's the actual problem.
